### DISCUSSION TITLE
`init_logit_bias` for SigLIP

### DISCUSSION THREAD

**Discussion by EIFY** [mlfoundations/open_clip#687] [Q&A] [answered]
Perhaps more of a question about SigLIP itself than about the open_clip implementation in particular, but I wonder if `init_logit_bias` should be initialized to `np.log(p / (1 - p))` instead where `p = 1/batch_size` to make sure that the model starts out at the loss minimum when the embeddings from the vision & text towers are just random noise? @lucasb-eyer
https://github.com/mlfoundations/open_clip/blob/91b7b512d26899e8b660414daea509d5bda38aec/src/training/main.py#L221
If this is the right initialization, `init_logit_bias=-10` would be the ideal value for batch size 22027. If I read [the paper](https://arxiv.org/abs/2303.15343) correctly, both the specific parameter sweep for `init_logit_bias` (batch size 8k) and the strongest results (batch size 16k-32k) used BS within an order of magnitude. I have tried training some toy models and it seems that for batch size 128/256 `init_logit_bias=np.log(p / (1 - p))` is no worse.

**Comment by rwightman** [comment]
I'll move to discussion instead of closing outright, but no sense keeping as an issue. Lucas might be more inclined to ans in the https://github.com/google-research/big_vision repo or on twitter...

**Comment by lucasb-eyer** [comment, marked as answer]
Hi, I don't actively monitor github, sorry.

I think this is an interesting idea that we didn't explore enough. However, from my exploration (albeit on a single batch-size) I'm pretty sure there needs to be a direct relationship between initialized temperature and bias. I'm not sure about how exactly, but am thinking about it in the background.
